src/modules/adaptors/Morpho/MorphoBlue/MorphoBlueCollateralAdaptor.sol

```diff
@@ -0,0 +1,230 @@
+     * @return Cellar's balance of provided collateral to specified MB market.
+     * @dev normal static call, thus msg.sender for sommelier focus is calling cellar
+     */
+    function balanceOf(bytes memory adaptorData) public view override returns (uint256) {
+        Id id = abi.decode(adaptorData, (Id));
+        return _userCollateralBalance(id, msg.sender);
+    }
+
+    /**
+     * @notice Returns collateral asset.
+     * @return The collateral asset in ERC20 type.
+     */
+    function assetOf(bytes memory adaptorData) public view override returns (ERC20) {
+        Id id = abi.decode(adaptorData, (Id));
+        MarketParams memory market = morphoBlue.idToMarketParams(id);
+        return ERC20(market.collateralToken);
+    }
+
+    /**
+     * @notice This adaptor returns collateral, and not debt.
+     * @return Whether or not this position is a debt position
+     */
+    function isDebt() public pure override returns (bool) {
+        return false;
+    }
+
+    //============================================ Strategist Functions ===========================================
+
+    /**
+     * @notice Allows strategists to add collateral to the respective cellar position on specified MB Market, enabling borrowing.
+     * @param _id encoded bytes32 MB id that represents the MB market for this position.
+     * @param _collateralToDeposit The amount of `collateralToken` to add to specified MB market position.
+     */
+    function addCollateral(Id _id, uint256 _collateralToDeposit) public {
+        _validateMBMarket(_id);
+        MarketParams memory market = morphoBlue.idToMarketParams(_id);
+        ERC20 collateralToken = ERC20(market.collateralToken);
+
+        uint256 amountToDeposit = _maxAvailable(collateralToken, _collateralToDeposit);
+        collateralToken.safeApprove(address(morphoBlue), amountToDeposit);
+
+        _addCollateral(market, amountToDeposit);
+
+        // Zero out approvals if necessary.
+        _revokeExternalApproval(collateralToken, address(morphoBlue));
+    }
+
+    /**
+     * @notice Allows strategists to remove collateral from the respective cellar position on specified MB Market.
+     * @param _id encoded bytes32 MB id that represents the MB market for this position.
+     * @param _collateralAmount The amount of collateral to remove from specified MB market position.
+     */
+    function removeCollateral(Id _id, uint256 _collateralAmount) public {
+        _validateMBMarket(_id);
+        MarketParams memory market = morphoBlue.idToMarketParams(_id);
+
+        _accrueInterest(market);
+        if (_collateralAmount == type(uint256).max) {
+            _collateralAmount = _userCollateralBalance(_id, address(this));
+        }
+
+        // remove collateral
+        _removeCollateral(market, _collateralAmount);
+
+        // TODO - might want to check the market to see if it even has a LLTV. If it doesn't, I guess no liquidations can occur?
```

Comment:
So if it doesn't have an LLTV, then we can not take on any borrows. So I think that it is okay to remove this check, because we will only add a 0 LLTV market to the registry, if we want to use them. But for now I dont see any use of them for us.

Reply:
sg2m!